### PR TITLE
Add 7002 ns support

### PR DESCRIPTION
**Pull request by krish2718**

### CODE REVIEW

**Review by sachinthegreen — APPROVED** [review]

File: samples/wifi/shell/overlay-tfm.conf [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+CONFIG_NET_RX_STACK_SIZE=2048
+
+# printing of scan results puts pressure on queues in new locking
+# design in net_mgmt. So, use a typical number of networks in a crowded
+# environment as the queue depth.
+CONFIG_NET_MGMT_EVENT_QUEUE_SIZE=10
+
+CONFIG_BUILD_WITH_TFM=y
+CONFIG_TFM_PROFILE_TYPE_NOT_SET=y
+CONFIG_TFM_BL2=n
```
Comment (joerchan) [inline samples/wifi/shell/overlay-tfm.conf:31]: Not needed to set this, BL2 is always disabled in NCS.

File: samples/wifi/shell/overlay-tfm.conf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+CONFIG_ENTROPY_PSA_CRYPTO_RNG=y
```
Comment (joerchan) [inline samples/wifi/shell/overlay-tfm.conf:32]: This should come from DTS file instead.

File: samples/wifi/shell/overlay-tfm.conf [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+CONFIG_NET_TX_STACK_SIZE=2048
+CONFIG_NET_RX_STACK_SIZE=2048
+
+# printing of scan results puts pressure on queues in new locking
+# design in net_mgmt. So, use a typical number of networks in a crowded
+# environment as the queue depth.
+CONFIG_NET_MGMT_EVENT_QUEUE_SIZE=10
+
+CONFIG_BUILD_WITH_TFM=y
```
Comment (joerchan) [inline samples/wifi/shell/overlay-tfm.conf:29]: This should come from defconfig for the board.

**Review by joerchan — CHANGES_REQUESTED** [review]

**Comment by joerchan** [comment]
Which board is this built for, I can give better feedback if you show me an example of something you want to build.

**Comment by krish2718** [comment]
> Which board is this built for, I can give better feedback if you show me an example of something you want to build.

`nrf7002dk_nrf5340_cpuapp_ns` this is the board that this PR adds support. and you can use `samples/wifi/shell` with `overlay-tfm.conf`

File: samples/wifi/shell/overlay-tfm.conf [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+CONFIG_TFM_PROFILE_TYPE_NOT_SET=y
+CONFIG_TFM_BL2=n
+CONFIG_ENTROPY_PSA_CRYPTO_RNG=y
+CONFIG_MBEDTLS_PSA_CRYPTO_C=y
+CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG=y
```
Comment (joerchan) [inline samples/wifi/shell/overlay-tfm.conf:34]: This ends up as disabled.

Conflicting with this?
```
# Needed for internal entropy
config MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG
    default n if WPA_SUPP_CRYPTO
```

Comment (krish2718) [reply samples/wifi/shell/overlay-tfm.conf:34]: Yeah, removed this now.

File: samples/wifi/shell/overlay-tfm.conf (hunk 3 above)
Comment (krish2718) [reply samples/wifi/shell/overlay-tfm.conf:29]: Done.

File: samples/wifi/shell/overlay-tfm.conf (hunk 2 above)
Comment (krish2718) [reply samples/wifi/shell/overlay-tfm.conf:32]: Done

File: samples/wifi/shell/overlay-tfm.conf (hunk 1 above)
Comment (krish2718) [reply samples/wifi/shell/overlay-tfm.conf:31]: Done

**Comment by joerchan** [comment]
> > Which board is this built for, I can give better feedback if you show me an example of something you want to build.
> 
> `nrf7002dk_nrf5340_cpuapp_ns` this is the board that this PR adds support. and you can use `samples/wifi/shell` with `overlay-tfm.conf`

I tried to build this, but I got a lot of build errors, and there are also build errors in CI.

**Comment by krish2718** [comment]
> > > Which board is this built for, I can give better feedback if you show me an example of something you want to build.
> > 
> > 
> > `nrf7002dk_nrf5340_cpuapp_ns` this is the board that this PR adds support. and you can use `samples/wifi/shell` with `overlay-tfm.conf`
> 
> I tried to build this, but I got a lot of build errors, and there are also build errors in CI.

I have fixed these once I test all combinations, will push new version.

**Review by joerchan — APPROVED** [review]

**Pull request merged**